KAFKA-16159: MINOR - Correcting logs to debug telemetry reporter

apoorvmittal10 opened the pull request:
Removed debug log as next time to update runs in poll loop and excessive logging happens.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

apoorvmittal10 committed:
KAFKA-16159: MINOR - Removed debug log

ex172000 commented on clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java:
```diff
@@ -362,7 +362,6 @@ public long timeToNextUpdate(long requestTimeoutMs) {
                     throw new IllegalStateException("Unknown telemetry state: " + localState);
             }
 
-            log.debug("For telemetry state {}, returning the value {} ms; {}", localState, timeMs, msg);
```
Do we want to emit some metrics instead so users can still get a sense of the state?

apoorvmittal10 replied:
I think emitting metrics would have been helpful if we want to derive some meaningful information about the running reporter, but here the log was helpful to see if reporter is working correctly while testing/debugging. Rather than removing it, I moved it to trace so still there is a way to know what's the state of reporter if we need to debug application.

philipnee commented:
Thanks @apoorvmittal10 - Could you elaborate on the purpose of this log line? how important is it for the user to know about the "next update time"?  I wonder if we could just log it when update is happening.

apoorvmittal10 committed:
Updated log level and added additional debug log

apoorvmittal10 commented:
> Thanks @apoorvmittal10 - Could you elaborate on the purpose of this log line? how important is it for the user to know about the "next update time"? I wonder if we could just log it when update is happening.

@philipnee The purpose of the log line was to debug if reporter is actually working and what's the time remaining while testing (was helpful in development). Rather than completely removing the line, I have moved it to trace so there is still some way for debugging later.

I have added another debug log line which will only be logged when a telemetry request is created in accordance with push interval time interval so that ll be minimal and will still let developer an idea if telemetry is working and in which state (get subscription or push telemetry).

philipnee commented on clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java:
```diff
@@ -379,6 +379,7 @@ public Optional<AbstractRequest.Builder<?>> createRequest() {
                 lock.readLock().unlock();
             }
 
+            log.debug("Creating telemetry request. Telemetry state: {}", localState);
```
it would be useful to know the type of the request client is sending: either it is a subscriptionRequest or push request.  I guess we can derive it from the current state, but state is rather an internal thing so that the person looking at it might not know what it means. my suggestion is to be explicit about "we are sending a push request" or "we are sending a subscription request".

apoorvmittal10 replied:
Hmmm, I don't have an opinion there, either seems fine to me. Your wish is my command :). Done.

philipnee commented on clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java:
```diff
@@ -362,7 +362,7 @@ public long timeToNextUpdate(long requestTimeoutMs) {
             }
 
-            log.debug("For telemetry state {}, returning the value {} ms; {}", localState, timeMs, msg);
+            log.trace("For telemetry state {}, returning the value {} ms; {}", localState, timeMs, msg);
```
Do you think it makes sense to rephrase the log? For the `msg` can we explicitly say:

...client will wait for {}ms before submitting the next...

philipnee replied:
so that we can remove the "returning the value 224678 ms" - it was a bit hard to understand what it actually mean.

apoorvmittal10 replied:
I don't think it will make much difference, seems fine to me. Writing the suggestion will involve changing `msg` in each block. I think it's fine the way it is :)

apoorvmittal10 committed:
Addressing review comment

philipnee reviewed (approved):
Thanks for the changes @apoorvmittal10 - cc @mjsax

mjsax commented:
Thanks @apoorvmittal10!

Merged to `trunk` and cherry-picked to `3.7` branch.
